docs/models/indexes.md

```diff
@@ -0,0 +1,338 @@
+
+class User extends Model {
+  @Attribute(DataTypes.STRING)
+  @NotNull
+  // highlight-next-line
+  @Index({ name: 'firstName-lastName' })
+  declare firstName: string;
+
+  @Attribute(DataTypes.STRING)
+  @NotNull
+  // highlight-start
+  @Index({ 
+    name: 'firstName-lastName',
+    // note: collations vary greatly between databases, this is just an example.
+    // You should check your database's documentation on collations for more information.
+    attribute: { collate: 'case_insensitive' },
+  })
+  // highlight-end
+  declare lastName: string;
+}
+```
+
+:::caution
+
+While it's possible to use options other than `name` and `attribute` in multi-column indexes that use the `@Index` decorator, the practice is discouraged as 
+you would have to repeat the same options on each attribute. Use [`createIndexDecorator`](#complex-multi-column-indexes) for that use case instead.
+
+:::
+
+## Complex Multi-column indexes
+
+The `createIndexDecorator` API is a convenient way to define complex multi-column indexes, without having to repeat the same options on each attribute.
+
+Without `createIndexDecorator`, a complex multi-column index would look like this:
+
+```ts
+import { Model, DataTypes } from '@sequelize/core';
+import { Index, Attribute, NotNull } from '@sequelize/core/decorators-legacy';
+
+class User extends Model {
+  @Attribute(DataTypes.STRING)
+  @NotNull
+  @Index({
+    // error-start
+    name: 'firstName-lastName', 
+    type: 'fulltext',
+    concurrently: true,
+    // error-end
+  })
+  declare firstName: string;
+
+  @Attribute(DataTypes.STRING)
+  @NotNull
+  @Index({
+    // error-start
+    name: 'firstName-lastName',
+    type: 'fulltext',
+    concurrently: true,
+    // error-end
+    attribute: { collate: 'case_insensitive' },
+  })
+  declare lastName: string;
+}
+```
+
+Notice how we had to repeat the same options on each attribute. This can become verbose very quickly. 
+With `createIndexDecorator`, you only have to specify the options once:
+
+```ts
+const NameIndex = createIndexDecorator('NameIndex', {
+  // success-start
+  name: 'firstName-lastName',
+  type: 'fulltext',
+  concurrently: true,
+  // success-end
+});
+
+class User extends Model {
+  @Attribute(DataTypes.STRING)
+  // highlight-next-line
+  @NameIndex
+  declare firstName: string;
+
+  @Attribute(DataTypes.STRING)
+  // highlight-start
+  // Attribute-specific options are still specified on the attribute:
+  @NameIndex({ collate: 'case_insensitive' })
+  // highlight-end
+  declare lastName: string;
+}
+```
+
+## Alternative Syntax (`@Table`)
+
+Another way to define an index is to use the `@Table` decorator on your model. This decorator accepts an `indexes` option, which is an array of index definitions.
+
+This option is able to define any and all indexes that can be defined using either the `@Index` decorator, or the `createIndexDecorator` API, but can be more cumbersome to use. Here is an example:
```

Comment:
```suggestion
This option can define any index that can be defined using either the `@Index` decorator, or the `createIndexDecorator` API, but it can be more cumbersome to use. Here is an example:
```

Not 100% sure.